dart-impl/mpi/src/dart_team_group.c

```diff
@@ -47,19 +48,22 @@ dart_ret_t dart_group_destroy(
   dart_group_t *group)
 {
 
-  if (group == NULL || *group == NULL) {
```

Comment:
@devreal This check seems to be wrong and leads to failures in various use cases.
If I'm not mistaken, `*group == NULL` is a valid state.
Please review my changes in `dart_group_destroy`, they apparently fixed this function.